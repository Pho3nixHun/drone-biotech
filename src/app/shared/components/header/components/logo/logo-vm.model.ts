export interface LogoVM {
    imageSrc: string;
    altText: string;
}
