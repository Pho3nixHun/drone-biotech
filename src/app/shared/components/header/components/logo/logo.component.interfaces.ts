export interface LogoComponentInterfaces {
  imageSrc: string;
  routerLink: string | string[];
  altText: string;
}
