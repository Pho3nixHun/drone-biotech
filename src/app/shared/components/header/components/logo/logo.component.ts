import { Component, input } from '@angular/core';
import { LogoComponentInterfaces } from '@components/header/components/logo/logo.component.interfaces';

@Component({
  selector: 'app-logo',
  standalone: true,
  imports: [],
  templateUrl: './logo.component.html',
})
export class LogoComponent {
  vm = input<LogoComponentInterfaces | null>(null, { alias: 'vm' });
}
