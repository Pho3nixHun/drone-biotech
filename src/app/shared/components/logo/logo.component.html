<a href="">
  <img [src]="imageSrc" alt="logo" class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0" />
</a>
