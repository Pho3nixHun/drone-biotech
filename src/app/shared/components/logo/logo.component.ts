import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-logo',
  standalone: true,
  imports: [],
  templateUrl: './logo.component.html',
})
export class LogoComponent {
  @Input()
  imageSrc!: string;
}
