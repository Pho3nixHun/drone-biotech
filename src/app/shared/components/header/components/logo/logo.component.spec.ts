import { ComponentFixture, TestBed } from '@angular/core/testing';
import { LogoComponent } from './logo.component';

describe('LogoComponent', () => {
  let component: LogoComponent;
  let fixture: ComponentFixture<LogoComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [LogoComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(LogoComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.debugElement.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should get input data on the img and the anchor element', () => {
    fixture.componentRef.setInput('vm', { imageSrc: 'assets/phoenix.jpg', routerLink: 'link' });
    fixture.detectChanges();
    const imgElement: HTMLImageElement = compiled.querySelector('img')!;
    const anchorElement: HTMLAnchorElement = compiled.querySelector('a')!;
    expect(imgElement.getAttribute('src')).toBe('assets/phoenix.jpg'); 
    expect(anchorElement.getAttribute('href')).toBe('link');
  });
});
