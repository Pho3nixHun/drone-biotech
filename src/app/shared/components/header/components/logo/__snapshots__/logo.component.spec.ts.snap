// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoComponent should get input data on the img and the anchor element 1`] = `
<div
  id="root2"
  ng-version="18.2.7"
>
  <a
    href="link"
  >
    <img
      alt="logo"
      class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0"
      src="assets/phoenix.jpg"
    />
  </a>
  <!--container-->
</div>
`;

exports[`LogoComponent should not render the template when there is no VM provided 1`] = `
<div
  id="root1"
  ng-version="18.2.7"
>
  <!--container-->
</div>
`;
